plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java

```diff
@@ -112,7 +114,8 @@ public BigQueryQueryPageSource(
                 table,
                 client.getProjectId(),
                 ImmutableList.copyOf(columnHandles),
-                filter);
+                filter,
+                table.limit());
```

Comment:
We pass the tableHandle right ? Can we extract it from the handle itself in `buildSql`